ocs_ci/ocs/uninstall.py

```diff
@@ -212,14 +198,79 @@ def uninstall_ocs():
             .get("storageClassName")
         )
 
+    cleanup_policy = (
+        storage_cluster.get()
+        .get("metadata")
+        .get("annotations")
+        .get("uninstall.ocs.openshift.io/cleanup-policy")
+    )
+
     log.info("Deleting storageCluster object")
     storage_cluster.delete(resource_name=constants.DEFAULT_CLUSTERNAME)
 
-    log.info("Removing CRDs")
+    storage_node_list = get_labeled_nodes(constants.OPERATOR_NODE_LABEL)
+    if cleanup_policy == "delete":
+        log.info("Cleanup policy set to delete. checking cleanup pods")
+        cleanup_pods = [
+            pod for pod in get_all_pods() if "cluster-cleanup-job" in pod.name
+        ]
+        for pod in cleanup_pods:
+            while pod.get().get("status").get("phase") != "Succeeded":
+                log.info(f"waiting for cleanup pod {pod.name} to complete")
+                TimeoutSampler(timeout=10, sleep=30)
+            log.info(f"Cleanup pod {pod.name} completed successfully ")
+
+        # confirm var/lib/rook is deleted
+        log.info("Confirming OCS artifacts were deleted from nodes")
+        for node in storage_node_list:
+            assert not ocp_obj.exec_oc_debug_cmd(
+                node=node, cmd_list=["ls -l var/lib/rook"]
+            ), "OCS artificats were not deleted from nodes "
+    else:
+        log.info("Cleanup policy set to retain. skipping nodes cleanup")
+
+    log.info("Deleting openshift-storage namespace")
+    ocp_obj.delete_project(constants.OPENSHIFT_STORAGE_NAMESPACE)
+    ocp_obj.wait_for_delete(constants.OPENSHIFT_STORAGE_NAMESPACE)
+    switch_to_project(constants.DEFAULT_NAMESPACE)
+
+    # step 10: TODO remove crypto from nodes. how do I check if nodes are encrypted?
+    for node in storage_node_list:
+        log.info(f"removing encryption from {node}")
+        ocp_obj.exec_oc_debug_cmd(node=node, cmd_list=[])
+
+    if lso_sc is not None:
+        log.info("Removing LSO")
+        uninstall_lso(lso_sc)
+
+    log.info("deleting noobaa storage class")
+    noobaa_sc = ocp.OCP(kind=constants.STORAGECLASS)
+    noobaa_sc.delete(resource_name=constants.NOOBAA_SC)
+
+    log.info("Unlabeling storage nodes")
+    node_objs = ocp.OCP(kind=constants.NODE).get().get("items")
+    label_nodes(nodes=node_objs, label=constants.OPERATOR_NODE_LABEL[:-3] + "-")
+    label_nodes(nodes=node_objs, label=constants.TOPOLOGY_ROOK_LABEL + "-")
+
+    log.info("Removing taints from storage nodes")  # TODO not working
+    taint_nodes(nodes=node_objs, taint_label=constants.OCS_TAINT + "-")
+
+    log.info("Deleting remaining OCS PVs (if there are any)")
+    try:
+        rbd_pv = ocp.OCP(kind=constants.PV, resource_name="ocs-storagecluster-ceph-rbd")
+        fs_pv = ocp.OCP(kind=constants.PV, resource_name="ocs-storagecluster-cephfs")
+        rbd_pv.delete()
+        fs_pv.delete()
+        log.info("OCS PVs deleted")
+    except Exception as e:
```

Comment:
isn't `Exception` too wide here?

Reply:
Nope, this PV depends on the type of deployment and won't be there always